### PR TITLE
Remove distro hashing

### PR DESCRIPTION
**Pull request by hennevogel**
We render/display `Distribution`s from a hash because the `Distribution`s from the interconnect are from an XML.
This PR changes `Distribution`s to be fully based on records. Including the in-memory cache for `Distribution`s from remote.

### CODE REVIEW

**Comment by hennevogel** [comment]
Removed DNM label, #11087 was merged.

**Review by danidoni — DISMISSED** [review]

File: src/api/app/controllers/distributions_controller.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def index
 
   # GET /distributions/1234
   def show
-    @distribution = Distribution.find(params[:id]).to_hash
+    @distribution = Distribution.find(params[:id])
```
Comment (danidoni) [inline src/api/app/controllers/distributions_controller.rb:26]: If you move this into a `before_action` you can reuse it below and fix the code climate issues about `destroy` action too complex

**Comment by hennevogel** [comment]
:monocle_face: the codeclimate issue is about `RepositoriesController`

File: src/api/app/controllers/distributions_controller.rb (hunk 1 above)
Comment (hennevogel) [reply src/api/app/controllers/distributions_controller.rb:26]: :monocle_face: the codeclimate issue is about `RepositoriesController`

Comment (danidoni) [reply src/api/app/controllers/distributions_controller.rb:26]: So true! Forget it, then.

File: src/api/config/clock.rb [hunk 2]
```diff
@@ -48,6 +48,10 @@ module Clockwork
     BsRequest.delayed_auto_accept
   end
 
+  every(1.hour, 'refresh remote distros') do
+    FetchRemoteDistributionsJob.perform_later
```
Comment (vpereira) [inline src/api/config/clock.rb:52]: why this is necessary after this PR and it wasn't needed before it?

Comment (hennevogel) [reply src/api/config/clock.rb:52]: Before we stored the `distribution.xml` from the  interconnect projects in memcache. See

https://github.com/openSUSE/open-build-service/pull/11223/files#diff-66b82a7133958122ff9b8c0436aba2f4a3eef044f869b90c61209221afc288f4L54-L57

Comment (hennevogel) [reply src/api/config/clock.rb:52]: then converted XML to XMLHash in the `DistributionController`. Used that XMLHash in the XML views.

Which also means we where processing the Hash in `RepositoryController` when displaying them in the UI yadda yadda yadda. Hash overflow....

Comment (vpereira) [reply src/api/config/clock.rb:52]: > Before we stored the `distribution.xml` from the interconnect projects in memcache

Isn't this approach better? Why should we keep updating it every hour regardless of traffic or not?

Comment (hennevogel) [reply src/api/config/clock.rb:52]: > Isn't this approach better? Why should we keep updating it every hour regardless of traffic or not?

Better how?

**Review by vpereira — APPROVED** [review]

**Pull request merged**